packages/cli/src/PublicApi/v1/handlers/workflows/workflows.service.ts

```diff
@@ -91,8 +91,20 @@ export async function deleteWorkflow(workflow: WorkflowEntity): Promise<Workflow
 	return await Container.get(WorkflowRepository).remove(workflow);
 }
 
-export async function updateWorkflow(workflowId: string, updateData: WorkflowEntity) {
-	return await Container.get(WorkflowRepository).update(workflowId, updateData);
+export async function updateWorkflow(
+	workflowId: string,
+	updateData: WorkflowEntity,
+	projectId?: string,
+	role?: WorkflowSharingRole,
+) {
+	if (!projectId && !role) {
+		return await Container.get(WorkflowRepository).update(workflowId, updateData);
+	}
+
+	return await Db.transaction(async (tx) => {
+		await tx.update(WorkflowEntity, workflowId, updateData);
+		await tx.update(SharedWorkflow, { workflowId }, { projectId, role });
+	});
```

Comment:
Should probably call `EnterpriseWorkflowService.transferOne`, since more things need done to transfer a workflow